okhttp/src/main/java/okhttp3/internal/platform/Platform.java

```diff
@@ -133,6 +133,20 @@ public boolean isCleartextTrafficPermitted(String hostname) {
     return true;
   }
 
+  public Object captureAllocationSite() {
+    if (!logger.isLoggable(Level.FINE)) return null;
```

Comment:
Use the log level for non-Android to detect whether to capture the allocation site.